### PR TITLE
ZIO 2.0.0-RC5

### PR DESCRIPTION
**Pull request by vigoo**

### CODE REVIEW

File: src/main/scala/zio/kafka/consumer/Consumer.scala [hunk 1]
```diff
@@ -193,21 +193,40 @@ object Consumer {
...
-          _.map { case (tp, partition) =>
-            val partitionStream =
-              if (settings.perPartitionChunkPrefetch <= 0) partition
-              else partition.bufferChunks(settings.perPartitionChunkPrefetch)
-
-            tp -> partitionStream.mapChunksZIO(_.mapZIO(_.deserializeWith(keyDeserializer, valueDeserializer)))
-          }
+    ): Stream[Throwable, Chunk[(TopicPartition, ZStream[R, Throwable, CommittableRecord[K, V]])]] = {
+      val partitions = runloop.partitions
+      val stream =
```
Comment (svroonland) [inline src/main/scala/zio/kafka/consumer/Consumer.scala:198]: Any particular reason for this change?

**Review by svroonland — APPROVED** [review]
LGTM, assuming the build becomes green of course. One question about one of the changes

File: src/main/scala/zio/kafka/consumer/Consumer.scala (hunk 1 above)
Comment (vigoo) [reply src/main/scala/zio/kafka/consumer/Consumer.scala:198]: Yes, there is no `mapZIO` on `Queue` anymore so I basically mixed the old `Queue#mapZIO` and `ZStream#fromQueue` to achieve the same. Would be nice if there is a better solution :)

Comment (adamgfraser) [reply src/main/scala/zio/kafka/consumer/Consumer.scala:198]: If you want to `Queue` is a trait now so you can implement a new queue that delegates to the old one if you really need that functionality. I think I had to do that one place in ZIO Stream.

File: src/main/scala/zio/kafka/consumer/internal/Runloop.scala [hunk 2]
```diff
@@ -409,8 +409,7 @@ private[consumer] final class Runloop(
       }
       .onError(cause => partitions.offer(Take.failCause(cause)))
       .unit
-      .toManaged
-      .fork
+      .forkScoped
```
Comment (adamgfraser) [inline src/main/scala/zio/kafka/consumer/internal/Runloop.scala:412]: This needs to be `forkScoped` because it was previously `ZManaged#fork`.

**Review by adamgfraser — APPROVED** [review]

**Pull request merged**